### ISSUE TITLE
NVDA is not announcing states of expanded or collapsed on button press

### ISSUE THREAD

**Issue by tgeczy** [nvaccess/nvda#8662]
### Steps to reproduce:
1. Open the URL https://www.microsoft.com/en-us/hololens/buyin
 2. Press the B key repeatedly until you arrive at the FAQ section with collapsed buttons.
3. Use enter or space to expand individual nodes and collapse them.

### Actual behavior:
NVDA will not read the given state. 

### Expected behavior:
NVDA should announce either "expanded" "collapsed" or repeat the full text of each button as they are altered. 
### System configuration:
Windos 10 1803 17134, Firefox 62 b18.

#### NVDA Installed/portable/running from source:
Installed
#### NVDA version:
2018.3
#### Windows version:
1803 17134
#### Name and version of other software in use when reproducing the issue:
Firefox 61 or above.
#### Other information about your system:
None applicable. 
### Other questions:
When trying to grab source from the DOM and copying the page to a local HTML, the issue does repro. However isolating the FAQ section reverts the behavior to normal. Perhaps the div nesting or some other layout difference is the root cause of this? 

#### Does the issue still occur after restarting your PC?
Yes

#### Have you tried any other versions of NVDA?
2018.2 - same result. Tried on other Firefox version also and was reproing in the stable release as well.
Running NVDA: alpha-15905,ec3eeee9

**Comment by zstanecic** [comment]
Hi,

Have you tried turning off the styles?

**Comment by tgeczy** [comment]
I just tried this. It will say "expanded" and "collapsed" but not repeating the button title. Still, this is better than it has been. The class I believe uses display:none. but it may actually work if a hidden one was used instead. Perhaps inline styling can create these kinds of issues.

**Comment by tgeczy** [comment]
I know I edited the version info here which can create confusions. Not to happy about having to edit it, but initially I discovered the issue on the alpha release and then decided to have a party of NVDA installs, working my way back to 2018.1 and 2018.2. It worked the same on all, but when writing the bug I as still on my 2018.2 copy and wrote that in the field. Hopefully this commeent will clarify the testing further.

**Comment by jscholes** [comment]
If you've isolated the problematic page section, as well as disabling the page styling, and both of them fixed the issue, it seems likely this is a page-specific problem for the page designers to solve.

**Comment by tgeczy** [comment]
Normally I would say so too. but I'm not so sure if they are able to isolate it down to that specific of a level. I have been trying to work with them and wanted to bring back something solid for them, but it almost feels like grasping at straws with accessibility. You can use Aria and design to being accessible but there are all of these unspoken rules which are screen reader specific. I think there's one here too, but I don't know exactly which style choice can create this type of impact. Once I can find that solution and it's a page change to be made, I'll close this bug. It may be that the issue is on Firefox or NVDA on how they follow a specific rule when interpreting page syntax.

**Comment by jcsteh** [comment]
[**@tgeczy**](https://github.com/tgeczy) commented on [Aug 23, 2018, 3:46 AM GMT+10](https://github.com/nvaccess/nvda/issues/8662#issue-353057347 "2018-08-22T17:46:43Z - Replied by Github Reply Comments"):
> 1. Open the URL https://www.microsoft.com/en-us/hololens/buyin

This URL reports page not found. However, this URL works: https://www.microsoft.com/en-us/hololens/buy

This is a duplicate of #5247. When the button expands or collapses, it changes in such a way that Firefox has to re-render the element completely. This causes the Firefox accessibility engine to re-create the button as if it were a new button, which then gets focus. Unfortunately, because of an NVDA issue, a new element which gets focus but occupies the same position where the cursor is already located does not get reported. Essentially, you're caught between two difficult-to-fix bugs, one in Firefox and one in NVDA.

Addressing another point:

[**@tgeczy**](https://github.com/tgeczy) commented on [Aug 23, 2018, 4:03 AM GMT+10](https://github.com/nvaccess/nvda/issues/8662#issuecomment-415124792 "2018-08-22T18:03:49Z - Replied by Github Reply Comments"):
> I just tried this. It will say "expanded" and "collapsed" but not repeating the button title. Still, this is better than it has been.

This is actually the expected behaviour. The button is already focused, so there's no need to repeat the name; we just report the change. Note that if the NVDA bug I mentioned above were fixed (but not the Firefox bug), it would report both the name, control type and state, which would be acceptable but still isn't the intended behaviour.

**Comment by tgeczy** [comment]
This is great news! Thank you so much for investigating the cause of the issue. I updated the URL as you referenced (perhaps through pasting something to added there) as it is the correct one. Very interesting to know about this problem, will be checking the other issue to see when this resolves on your end. Let me know if I should do anything with the state of this bug in the meantime as it does doop with that other one.
